### PR TITLE
added get history endpoint

### PR DESCRIPTION
**Pull request by RytisSapka**
Currently waiting for endpoint to reach prod before merging

### CODE REVIEW

File: src/reporting/clients/IExtractionClient.ts [hunk 1]
```diff
@@ -59,4 +59,33 @@ export interface IExtractionClient {
     accessToken: AccessToken,
     jobId: string
   ): Promise<ExtractionStatus>;
+
+  /**
+   * Gets Extractions of an IModel. This method returns the full list of Extractions.
+   * @param {string} iModelId Unique Identifier of the target IModel.
```
Comment (MindaugasButkus) [inline src/reporting/clients/IExtractionClient.ts:65]: Documentation is inconsistent with the methods above.
```suggestion
   * @param {string} iModelId The iModel Id.
```

File: src/reporting/clients/IExtractionClient.ts [hunk 2]
```diff
@@ -59,4 +59,33 @@ export interface IExtractionClient {
...
+  getExtractionHistory(
+    accessToken: AccessToken,
+    iModelId: string,
+    top?: number
+  ): Promise<Extraction[]>;
+
+  /**
+   * Gets an async paged iterator of Extractions for an IModel.
+   * This method returns an iterator which loads pages of extraction logs as it is being iterated over.
+   * @param {string} iModelId Unique Identifier of the target IModel.
```
Comment (MindaugasButkus) [inline src/reporting/clients/IExtractionClient.ts:80]:
```suggestion
   * @param {string} iModelId The iModel Id.
```

File: src/reporting/interfaces/ExtractionProcess.ts [hunk 3]
```diff
@@ -203,6 +223,54 @@ export interface ExtractionStatus {
...
 }
 
+/**
+ * List of Extractions.
+ * @export
+ * @interface ExtractionCollection
+ */
+export interface ExtractionCollection {
+  /**
+   * List of Extraction Logs.
```
Comment (MindaugasButkus) [inline src/reporting/interfaces/ExtractionProcess.ts:233]: These are not logs. They are just extractions. Logs have a separate type.

File: src/reporting/interfaces/ExtractionProcess.ts [hunk 4]
```diff
@@ -203,6 +223,54 @@ export interface ExtractionStatus {
...
+ */
+export interface Extraction {
+  /**
+   * Unique Identifier for the Extraction Run. Use this to check run status.
+   * @type {string}
+   * @memberof Extraction
+   */
+  jobId: ExtractorState;
+  /**
+   * Additional justification for the current state of the Extraction Run.
```
Comment (MindaugasButkus) [inline src/reporting/interfaces/ExtractionProcess.ts:260]: `startedOn` is not a justification.

File: src/test/ExtractionClient.test.ts [hunk 5]
```diff
@@ -141,4 +141,78 @@ describe("Extraction Client", () => {
...
+      },
+    };
+    const returns2 = {
+      extractions: [3, 4],
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      _links: {
+        next: undefined,
+      },
+    };
+    fetchStub.withArgs("https://api.bentley.com/insights/reporting/datasources/iModels/-/extraction/history", "pass").resolves(returns1)
```
Comment (MindaugasButkus) [inline src/test/ExtractionClient.test.ts:162]: Please avoid using dashes in place of guids. It's hard to understand what they mean.

Comment (MindaugasButkus) [reply src/test/ExtractionClient.test.ts:162]: You pass 2 dashes into `getExtractionHistory` method - which dash is added to the url?

File: src/reporting/interfaces/ExtractionProcess.ts (hunk 4 above)
Comment (RytisSapka) [reply src/reporting/interfaces/ExtractionProcess.ts]: Fixed comments (forgot about them before creating the PR, my bad).

File: src/test/ExtractionClient.test.ts (hunk 5 above)
Comment (RytisSapka) [reply src/test/ExtractionClient.test.ts]: Removed dashes for this test. Perhaps could fix all other tests in another PR.

**Review by MindaugasButkus — APPROVED** [review]

File: src/reporting/clients/IExtractionClient.ts [hunk 6]
```diff
@@ -59,4 +59,33 @@ export interface IExtractionClient {
...
+
+  /**
+   * Gets Extractions of an IModel. This method returns the full list of Extractions.
+   * @param {string} iModelId The iModel Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`.
+   * @param {number} top The number of entities to load per page.
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/get-extraction-history/
+   */
+  getExtractionHistory(
```
Comment (MindaugasButkus) [inline src/reporting/clients/IExtractionClient.ts:71]: Bump patch version in the package.json file so that we can release this package.

**Pull request merged**